src/include/mpir_progress.h

```diff
@@ -0,0 +1,16 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2020 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+#ifndef MPIR_PROGRESS_H_INCLUDED
+#define MPIR_PROGRESS_H_INCLUDED
+
+int MPIR_Progress_hooks_exec(int *made_progress);
+int MPIR_Progress_register(int (*progress_fn) (int *), int *id);
+int MPIR_Progress_deregister(int id);
+int MPIR_Progress_activate(int id);
+int MPIR_Progress_deactivate(int id);
+
```

Comment:
How about naming them consistently with `MPIR_Progress_hooks_` prefix? "Progress" confuses with the main progress in `MPI_Wait/Test`.

Reply:
Fixed the naming and also rebased on to the latest `master`.